lxd/storage_volumes.go

```diff
@@ -1924,10 +1924,6 @@ func storagePoolVolumeTypePostMove(s *state.State, r *http.Request, poolName str
 			return err
 		}
 
-		revert.Add(func() {
```

Comment:
the downside with this approach is that now if any subsequent function fails, like `newPool.CreateCustomVolumeFromCopy` or `pool.DeleteCustomVolume` the updated volume users wont get reverted.

Suggest instead that you keep your updated implementation, but update `storagePoolVolumeUpdateUsers` to return a `revert.Hook` (like `clusterInitMember`).

That way the contract can be that `storagePoolVolumeUpdateUsers` will deal with its own reverts if it returns a non-nil error, but if it returns a nil error then it will also return a revert.Hook that can be added to the outer reverter to be undone if a subsequent step fails.

Reply:
and `storagePoolVolumeTypePostRename` can then benefit from that approach to and use an outer reverter too.